main.go

```diff
@@ -67,14 +71,62 @@ func init() {
 	// +kubebuilder:scaffold:scheme
 }
 
+func newPromMonitor(errCh chan error) *promMonitor {
+	monitor := &promMonitor{
+		errCh: errCh,
+	}
+
+	return monitor
+}
+
+type promMonitor struct {
+	errCh chan error
+}
+
+func (m *promMonitor) hasMigrationErr() error {
+	select {
+	case err := <-m.errCh:
+		if err != nil {
+			return err
+		}
+	default:
+	}
+
+	return nil
+}
+
+func (m *promMonitor) startMonitor() {
+	for {
+		time.Sleep(400 * time.Millisecond)
```

Comment:
Is this sleep necessary?

Reply:
nope, and like @kwiesmueller  pointed out, it is all useless :) so now it is gone :)